HOApp/ViewModel/POEntryViewModel.cs

```diff
@@ -1,106 +1,129 @@
+                    _selectedStore = value;
+                    RaisePropertyChanged();
+                }
             }
         }
 
-        public POEntryViewModel() : base() { }
+        public IReadOnlyList<ProductVM> SelectableProducts => _selectableProducts;
+
+        public ProductVM SelectedProduct { get; set; }
 
         protected async override void GetData(string filter = "")
         {
-            //TODO: The throbber in each screen works, but it's covered in the UI
             ThrobberVisible = Visibility.Visible;
 
-            //Supplier dropdown TODO: Should only look at active suppliers
-            ObservableCollection<SupplierVM> _suppliers = new ObservableCollection<SupplierVM>();
-            var suppliers = await (from s in db.Suppliers
-                                   orderby s.Description
-                                   select s).ToListAsync();
-            foreach (Supplier supplier in suppliers)
+            using (var db = new RetailDbContext())
             {
-                _suppliers.Add(new SupplierVM { IsNew = false, TheEntity = supplier });
-            }
-            Suppliers = _suppliers;
-            RaisePropertyChanged("Suppliers");
-
-            //Store dropdown TODO: Should only look at active stores
-            ObservableCollection<StoreVM> _stores = new ObservableCollection<StoreVM>();
-            var stores = await (from s in db.Stores
-                                   orderby s.Description
-                                   select s).ToListAsync();
-            foreach (Store store in stores)
-            {
-                _stores.Add(new StoreVM { IsNew = false, TheEntity = store });
-            }
-            Stores = _stores;
-            RaisePropertyChanged("Stores");
-
-            //Product dropdown TODO: Should only look at products with an active status
-            ObservableCollection<ProductVM> _products = new ObservableCollection<ProductVM>();
-            var products = await (from s in db.Products
-                                orderby s.Description
-                                select s).ToListAsync();
-            foreach (Product product in products)
-            {
-                _products.Add(new ProductVM { IsNew = false, TheEntity = product });
+                Suppliers = new ObservableCollection<SupplierVM>
+                (
+                    await db.Suppliers
+                    .OrderBy(n => n.Description)
+                    .Select(n => new SupplierVM { IsNew = false, TheEntity = n })
+                    .ToListAsync()
+                    .ConfigureAwait(false)
+                );
+
+                Stores = new ObservableCollection<StoreVM>
+                (
+                    await db.Stores
+                    .OrderBy(n => n.Description)
+                    .Select(n => new StoreVM { IsNew = false, TheEntity = n })
+                    .ToListAsync()
+                    .ConfigureAwait(false)
+                );

```

Comment:
Fluent Linq(`.Where().Select()`) preferred over query comprehension(`from x in a`) in most cases as it more clearly defines a chain of operations.